### PR TITLE
Add DialDtmfDialog

### PR DESCRIPTION
**Pull request by cbeckr**
Adds a simple dialog to dial number sequences:

To enable, add ‘dtmf’ to TOOLBAR_BUTTONS in interface_config.js

Solves https://github.com/jitsi/jitsi-meet/issues/6332

8x8 CLA completed 2021-03-26

### CODE REVIEW

**Comment by mwllgr** [comment]
Any news on the feature?

File: lang/main-de.json [hunk 1]
```diff
@@ -324,6 +324,13 @@
     "documentSharing": {
         "title": "Freigegebenes Dokument"
     },
+    "dialDtmf": {
```
Comment (muscat1) [inline lang/main-de.json:327]: Not sure if we're supposed to add individual translations like this. Don't we usually have someone that updates the files in bulk, @saghul ?

File: react/features/dtmf/components/DialDtmfDialog.js [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    /**
+     * The currently entered pause duration (ms).
+     */
+    pauseDuration: number
+};
+
+/**
+ * A dialog that allows to dial a DTMF number sequence.
+ */
+class DialDtmfDialog extends Component<Props, State> {
```
Comment (muscat1) [inline react/features/dtmf/components/DialDtmfDialog.js:45]: This doesn't seem like too complex of a component, would it be possible to refactor it into a function component? We're trying to move towards a more functional style, especially in simpler use cases.

File: react/features/dtmf/components/DialDtmfDialog.js [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    _onSubmit() {
+        const { numbers, toneDuration, pauseDuration } = this.state;
+
+        APP.store.dispatch(sendTones(numbers, toneDuration, pauseDuration));
+
+        return true;
+    }
+}
+
+const mapStateToProps = () => {
```
Comment (muscat1) [inline react/features/dtmf/components/DialDtmfDialog.js:198]: This shouldn't be necessary, you're not using any props from the redux store.

File: react/features/dtmf/components/DialDtmfDialog.js [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    return {
+    };
+};
+
+export default translate(connect(mapStateToProps)(DialDtmfDialog));
```
Comment (muscat1) [inline react/features/dtmf/components/DialDtmfDialog.js:203]: The `connect` here isn't necessary either, same as above.

File: react/features/toolbox/components/web/Toolbox.js [hunk 5]
```diff
@@ -1058,7 +1087,14 @@ class Toolbox extends Component<Props> {
             this._shouldShowButton('help')
                 && <HelpButton
                     key = 'help'
-                    showLabel = { true } />
+                    showLabel = { true } />,
+            this._shouldShowButton('dtmf')
+                && <OverflowMenuItem
+                    accessibilityLabel = { t('toolbar.accessibilityLabel.dtmf') }
+                    icon = { IconTileView }
```
Comment (muscat1) [inline react/features/toolbox/components/web/Toolbox.js:1094]: Should probably use a different icon here.

**Review by muscat1 — CHANGES_REQUESTED** [review]
Left a few comments. Maybe @saghul can also take a look, in case I might've missed something.

File: lang/main-de.json (hunk 1 above)
Comment (damencho) [reply lang/main-de.json:327]: This is fine. We still accept translations as PRs, once we move to some system for translations, we will accept in PRs only main.json changes.

**Comment by c-cal** [comment]
Is it possible to help make this PR move forward without having to create a new one? This is a much needed feature for us.

**Comment by cbeckr** [comment]
@c-cal I'll try to follow up here over the next week or so.

**Comment by saghul** [comment]
https://github.com/jitsi/jitsi-meet/issues/6332#issuecomment-1108293273

**Pull request closed without merging**